vim-patch:9.0.0403: 'equalalways' may be off when 'laststatus' is zero

luukvbaal opened the pull request:
Problem:    'equalalways' may be off when 'laststatus' is zero.
Solution:   call last_status() before win_equal(). (Luuk van Baal,
            closes https://github.com/vim/vim/pull/11070)
https://github.com/vim/vim/commit/fd7e60a33ddd83a82da4eb6267f1c12fa926f32b

No idea how you guys handle porting vim patches but I thought I would open this PR since I'm it's author and am primarily a neovim user.

zeertzjq commented on src/nvim/testdir/test_window_cmd.vim:
```diff
@@ -1464,4 +1464,18 @@ func Test_win_move_statusline()
+  set splitbelow&
+  set laststatus&
+endfunc
+
```
missing blank line
```suggestion


```

zeertzjq replied:
`test_put.vim` does. It's better to avoid divergence from Vim in test files.

zeertzjq commented on src/nvim/window.c:
```diff
@@ -2849,6 +2849,11 @@ int win_close(win_T *win, bool free_buf, bool force)
   }
 
+  // If last window has a status line now and we don't want one,
+	// remove the status line. Do this before win_equal(), because
+	// it may change the height of a window.
```
```suggestion
  // remove the status line. Do this before win_equal(), because
  // it may change the height of a window.
```

clason replied:
Or just `make format`

luukvbaal replied:
Done, sorry for the mess.

luukvbaal committed:
vim-patch:9.0.0403: 'equalalways' may be off when 'laststatus' is zero
Problem:    'equalalways' may be off when 'laststatus' is zero.
Solution:   call last_status() before win_equal(). (Luuk van Baal,
            closes https://github.com/vim/vim/pull/11070)
https://github.com/vim/vim/commit/fd7e60a33ddd83a82da4eb6267f1c12fa926f32b
